internal/internal_poller_autoscaler.go

```diff
@@ -60,31 +62,40 @@ type (
 		// This avoids unnecessary usage of CompareAndSwap
 		atomicBits *atomic.Uint64
 	}
+
+	pollerAutoScalerOptions struct {
+		Enabled           bool
+		InitCount         int
+		MinCount          int
+		MaxCount          int
+		Cooldown          time.Duration
+		DryRun            bool
+		TargetUtilization float64
+	}
 )
 
 func newPollerScaler(
-	initialPollerCount int,
-	minPollerCount int,
-	maxPollerCount int,
-	targetMilliUsage uint64,
-	isDryRun bool,
-	cooldownTime time.Duration,
+	options pollerAutoScalerOptions,
 	logger *zap.Logger,
 	hooks ...func()) *pollerAutoScaler {
 	ctx, cancel := context.WithCancel(context.Background())
```

Comment:
is this a sign we should be passing in a context?  e.g. to propagate shutdown information maybe.